.ci/check_python_dists.sh

```diff
@@ -17,4 +18,12 @@ if { test "${TASK}" = "bdist" || test "${METHOD}" = "wheel"; }; then
     check-wheel-contents ${DIST_DIR}/*.whl || exit -1
 fi
 
+echo "pydistcheck..."
+pydistcheck \
+    --inspect \
```

Comment:
This argument `--inspect` prints a summary like the following:

```text
----- package inspection summary -----
file size
  * compressed size: 1.3M
  * uncompressed size: 4.3M
  * compression space saving: 69.5%
contents
  * directories: 0
  * files: 16 (1 compiled)
size by extension
  * .so - 3963.7K (90.5%)
  * .py - 396.7K (9.1%)
  * no-extension - 17.6K (0.4%)
  * .txt - 0.0K (0.0%)
  * .typed - 0.0K (0.0%)
------------ check results -----------
errors found while checking: 0

```

([macOS bdist build link](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=14559&view=logs&j=52ea7a35-6bb0-5680-4def-de74fd7388e2&t=3a5da3c1-f658-52cd-3d24-edc674c4d20e))

Which I think would have been useful in situations like this: https://github.com/microsoft/LightGBM/pull/5252#pullrequestreview-1113112842